### DISCUSSION TITLE
Extra Perception Warning

### DISCUSSION THREAD

**Discussion by nmalevich** [pointfreeco/swift-composable-architecture#3069] [General]
### Description

Show unnecessary perception tracking runtime warning. You can check that I capture all store accesses.
<img width="992" alt="Снимок экрана 2024-05-10 в 13 56 06" src="https://github.com/pointfreeco/swift-composable-architecture/assets/147712544/90a02fe1-502d-4343-a7d5-eb409d756164">

### Checklist

- [ ] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

Not show warning

### Actual behavior

Show warning

### Steps to reproduce

1. Run
2. Tap on "Tap" button
3. You see runtime warning
[BadPerceptibleWarning.zip](https://github.com/pointfreeco/swift-composable-architecture/files/15274110/BadPerceptibleWarning.zip)

### The Composable Architecture version information

1.10.3

### Destination operating system

iOS 17.4.1

### Xcode version information

15.3

### Swift Compiler version information

```shell
swift-driver version: 1.90.11.1 Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)
```

**Comment by mbrandonw** [comment]
Hi @nmalevich, the problem is in `ActivityIndicatorView`. If you comment that out you will see it works. And the reason there is a problem is because `ActivityIndicatorView` takes a binding and it accesses the state in its `body`, and of course there is no `WithPerceptionTracking` over there.

Really it's not clear why `ActivityIndicatorView` even takes a binding. It never writes to the binding. It would be better if it took a simple `Bool`, and then you would not have this problem.

Since this is not an issue with the library I am going to convert it to a discussion. Feel free to continue the conversation over there.

**Comment by nmalevich** [comment]
Hi @mbrandonw
Thank you for your answer!

Yes, I got that reason of problem is ActivityIndicatorView and I agree with you about unnecessary taking binding in that case. But will not this problem appear if I just use every third part view? So I think it is problem if I can't use third part views with TCA.

Thank you one more time for your help. I will be grateful for your answer
